packages/shared_preferences/shared_preferences_android/android/src/test/java/io/flutter/plugins/sharedpreferences/SharedPreferencesTest.java

```diff
@@ -4,12 +4,270 @@
+    }
+  }
+
+  SharedPreferencesPlugin plugin;
+
+  @Mock BinaryMessenger mockMessenger;
+  @Mock FlutterPlugin.FlutterPluginBinding flutterPluginBinding;
+
+  @Before
+  public void before() {
+    Context context = Mockito.mock(Context.class);
+    SharedPreferences sharedPrefs = new LocalSharedPreferences();
+
+    flutterPluginBinding = Mockito.mock(FlutterPlugin.FlutterPluginBinding.class);
+
+    Mockito.when(flutterPluginBinding.getBinaryMessenger()).thenReturn(mockMessenger);
+    Mockito.when(flutterPluginBinding.getApplicationContext()).thenReturn(context);
+    Mockito.when(context.getSharedPreferences(anyString(), anyInt())).thenReturn(sharedPrefs);
+
+    plugin = new SharedPreferencesPlugin(new ListEncoder());
+    plugin.onAttachedToEngine(flutterPluginBinding);
+  }
+
+  private static final Map<String, Object> data = new HashMap<>();
+
+  static {
+    data.put("Language", "Java");
+    data.put("Counter", 0L);
+    data.put("Pie", 3.14);
+    data.put("Names", Arrays.asList("Flutter", "Dart"));
+    data.put("NewToFlutter", false);
+    data.put("flutter.Language", "Java");
+    data.put("flutter.Counter", 0L);
+    data.put("flutter.Pie", 3.14);
+    data.put("flutter.Names", Arrays.asList("Flutter", "Dart"));
+    data.put("flutter.NewToFlutter", false);
+    data.put("prefix.Language", "Java");
+    data.put("prefix.Counter", 0L);
+    data.put("prefix.Pie", 3.14);
+    data.put("prefix.Names", Arrays.asList("Flutter", "Dart"));
+    data.put("prefix.NewToFlutter", false);
+  }
+
+  @Test
+  public void getAllWithPrefix() {
+    plugin.clearWithPrefix("");
+
+    assertEquals(plugin.getAllWithPrefix("").size(), 0);
+
+    addData();
+
+    Map<String, Object> flutterData = plugin.getAllWithPrefix("flutter.");
+
+    assertEquals(flutterData.size(), 5);
+    assertEquals(flutterData.get("flutter.Language"), "Java");
+    assertEquals(flutterData.get("flutter.Counter"), 0L);
+    assertEquals(flutterData.get("flutter.Pie"), 3.14);
+    assertEquals(flutterData.get("flutter.Names"), Arrays.asList("Flutter", "Dart"));
+    assertEquals(flutterData.get("flutter.NewToFlutter"), false);
+
+    Map<String, Object> allData = plugin.getAllWithPrefix("");
+
+    assertEquals(allData, data);
+  }
+
   @Test
-  public void initPluginDoesNotThrow() {
-    final SharedPreferencesPlugin plugin = new SharedPreferencesPlugin();
+  public void clearWithPrefix() {
+    plugin.clearWithPrefix("");
+
+    addData();
+
+    assertEquals(plugin.getAllWithPrefix("").size(), 15);
+
+    plugin.clearWithPrefix("flutter.");
+
+    assertEquals(plugin.getAllWithPrefix("").size(), 10);
+  }
+
+  @Test
+  public void testRemove() {
+    plugin.clearWithPrefix("");
+
+    plugin.setBool("isJava", true);
+    assert (plugin.getAllWithPrefix("").containsKey("isJava"));
+    plugin.remove("isJava");
+    assertFalse(plugin.getAllWithPrefix("").containsKey("isJava"));
+  }
+
```

Comment:
Currently you're only testing setting as a side-effect of other tests, which is fragile to refactoring. (E.g., what if `getAllWithPrefix` goes away in some future version? Someone shouldn't have to think about what tests they should *add* to remove code.) Each setter should have its own test method, even though that's currently somewhat redundant with other tests.